app/controllers/application_controller.rb

```diff
@@ -35,7 +35,14 @@ def analytics
   end
 
   def create_user_event(event_type, user = current_user)
-    Event.create(user_id: user.id, event_type: event_type)
+    Event.transaction do
+      Event.create!(user_id: user.id, event_type: event_type)
+      # Keep total number of events per user under MAX_EVENTS_PER_USER
+      events = Event.where(user_id: user.id).order('created_at ASC')
+      if events.count > Event::MAX_EVENTS_PER_USER
```

Comment:
+1 on rake task. I don't think it's a good idea to check the count on every request. Also, this is not the responsibility of the create_event method. Creating and purging are two different responsibilities.